storage/store/inmemory/unsynchronized/light_transaction_results.go

```diff
@@ -0,0 +1,110 @@
+package unsynchronized
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/store"
+)
+
+type LightTransactionResults struct {
+	// TODO: A mutexes aren't strictly necessary here since, by design, data is written only once
+	// before any future reads. However, we're keeping it temporarily during active development
+	// for safety and debugging purposes. It will be removed once the implementation is finalized.
+	store     map[string]flow.LightTransactionResult
+	storeLock sync.RWMutex
+
+	indexStore     map[string]flow.LightTransactionResult
+	indexStoreLock sync.RWMutex
+
+	blockStore     map[string][]flow.LightTransactionResult
+	blockStoreLock sync.RWMutex
+}
+
+func NewLightTransactionResults() *LightTransactionResults {
+	return &LightTransactionResults{
+		store:          make(map[string]flow.LightTransactionResult),
+		storeLock:      sync.RWMutex{},
+		indexStore:     make(map[string]flow.LightTransactionResult),
+		indexStoreLock: sync.RWMutex{},
+		blockStore:     make(map[string][]flow.LightTransactionResult),
+		blockStoreLock: sync.RWMutex{},
```

Comment:
nit: since the mutexes are stored by value, you don't need to explicitly initialize them
```suggestion
		store:          make(map[string]flow.LightTransactionResult),
		indexStore:     make(map[string]flow.LightTransactionResult),
		blockStore:     make(map[string][]flow.LightTransactionResult),
```